File: src/main/java/ru/ioleksiv/telegram/bot/core/controller/annotations/parser/filter/telegram/ChosenInlineResultFilterParser.java
package ru.ioleksiv.telegram.bot.core.controller.annotations.parser.filter.telegram;

import org.springframework.stereotype.Component;
import ru.ioleksiv.telegram.bot.api.annotations.filter.primitive.StringFilter;
import ru.ioleksiv.telegram.bot.api.annotations.filter.telegram.ChosenInlineResultFilter;
import ru.ioleksiv.telegram.bot.api.annotations.filter.telegram.LocationFilter;
import ru.ioleksiv.telegram.bot.api.annotations.filter.telegram.UserFilter;
import ru.ioleksiv.telegram.bot.api.model.objects.inline.ChosenInlineResult;
import ru.ioleksiv.telegram.bot.core.controller.annotations.parser.filter.FilterParser;
import ru.ioleksiv.telegram.bot.core.controller.annotations.parser.finder.Finder;
import ru.ioleksiv.telegram.bot.core.controller.handler.check.Validator;
import ru.ioleksiv.telegram.bot.core.controller.handler.check.impl.UnionExtractValidator;

import java.util.Optional;

@Component
public class ChosenInlineResultFilterParser implements FilterParser<ChosenInlineResultFilter, ChosenInlineResult> {

    @Override
    public Validator<ChosenInlineResult> createChecker(ChosenInlineResultFilter annotation, Finder finder) {
        UnionExtractValidator<ChosenInlineResult> unionExtractValidator = new UnionExtractValidator<>();

        UserFilter from = annotation.from();
        if (from.status().isActive()) {
            unionExtractValidator.add(in -> Optional.ofNullable(in.getFrom()), finder.find(from));
        }
        LocationFilter location = annotation.location();
        if (location.status().isActive()) {
            unionExtractValidator.add(in -> Optional.ofNullable(in.getLocation()), finder.find(location));
        }
        StringFilter inlineMessageId = annotation.inlineMessageId();
        if (inlineMessageId.status().isActive()) {
            unionExtractValidator.add(in -> Optional.ofNullable(in.getInlineMessageId()), finder.find(inlineMessageId));
        }
        StringFilter query = annotation.query();
        if (query.status().isActive()) {
            unionExtractValidator.add(in -> Optional.ofNullable(in.getQuery()), finder.find(query));
        }

        return unionExtractValidator;
    }

    @Override
    public Class<ChosenInlineResultFilter> getAnnotationClass() {
        return ChosenInlineResultFilter.class;
    }

}

File: core/src/main/kotlin/io/github/oleksivio/tl/kbot/core/core/controller/annotations/parser/filter/telegram/ChosenInlineResultFilterParser.kt
package io.github.oleksivio.tl.kbot.core.core.controller.annotations.parser.filter.telegram

import io.github.oleksivio.tl.kbot.core.annotations.filter.composite.ChosenInlineResultFilter
import io.github.oleksivio.tl.kbot.server.api.objects.inline.ChosenInlineResult
import io.github.oleksivio.tl.kbot.core.core.controller.annotations.parser.filter.FilterParser
import io.github.oleksivio.tl.kbot.core.core.controller.annotations.parser.finder.Finder
import io.github.oleksivio.tl.kbot.core.core.controller.handler.check.Validator
import io.github.oleksivio.tl.kbot.core.core.controller.handler.check.impl.UnionExtractValidator
 

  
class ChosenInlineResultFilterParser :
    FilterParser<ChosenInlineResultFilter, ChosenInlineResult> {

    override val parserAnnotationClass = ChosenInlineResultFilter::class

    override fun createChecker(annotation: ChosenInlineResultFilter, finder: Finder): Validator<ChosenInlineResult> {
        val unionExtractValidator =
            UnionExtractValidator<ChosenInlineResult>()

        annotation.validator
                .map { validatorName -> finder.find(validatorName, ChosenInlineResult::class) }
                .forEach { validator -> unionExtractValidator.add({ it }, validator) }

        val from = annotation.from
        if (from.status.isActive) {
            unionExtractValidator.add({ it.from }, finder.find(from))
        }
        val location = annotation.location
        if (location.status.isActive) {
            unionExtractValidator.add({ it.location }, finder.find(location))
        }
        val inlineMessageId = annotation.inlineMessageId
        if (inlineMessageId.status.isActive) {
            unionExtractValidator.add({ it.inlineMessageId }, finder.find(inlineMessageId))
        }
        val query = annotation.query
        if (query.status.isActive) {
            unionExtractValidator.add({ it.query }, finder.find(query))
        }

        return unionExtractValidator
    }

}

